include/swift/AST/Decl.h

```diff
@@ -1805,11 +1805,16 @@ class PatternBindingEntry {
   /// The initializer context used for this pattern binding entry.
   DeclContext *InitContext = nullptr;
 
+  /// The location of the equal '=' token.
+  SourceLoc EqualLoc;
+
   friend class PatternBindingInitializer;
 
 public:
-  PatternBindingEntry(Pattern *P, Expr *E, DeclContext *InitContext)
-    : ThePattern(P), InitAndFlags(E, {}), InitContext(InitContext) {}
+  PatternBindingEntry(Pattern *P, Expr *E, DeclContext *InitContext,
+                      SourceLoc EqualLoc)
```

Comment:
Usually, AST node constructors/factory methods receive parameters in source order. like:
```cpp
  PatternBindingEntry(Pattern *P, SourceLoc EqualLoc, Expr *E, DeclContext *InitContext)
```
Although I don't have strong opinion here, I think we should be consistent with other decls.

Reply:
Agreed, I'll go ahead and make that change. Thanks for the review!